addons/udes_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -1,5 +1,202 @@
-from odoo import models
+from odoo import models, fields, _, api
+from odoo.exceptions import UserError, ValidationError
+import logging
+
+_logger = logging.getLogger(__name__)
 
 
 class StockPickingBatch(models.Model):
     _inherit = "stock.picking.batch"
+
+    user_id = fields.Many2one(
+        "res.users",
+        states={
+            "draft": [("readonly", False)],
+            "waiting": [("readonly", False)],
+            "ready": [("readonly", False)],
+            "in_progress": [("readonly", False)],
+        },
+    )
+    # Note: state field has been found to recompute itself everytime it was accessed even with store=True
+    # lock_batch_state has been put in places to ensure correct behaviour.
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("waiting", "Waiting"),
+            ("ready", "Ready"),
+            ("in_progress", "Running"),
+            ("done", "Done"),
+            ("cancel", "Cancelled"),
+        ],
+        compute="_compute_state",
+        store=True,
+    )
+    picking_ids = fields.One2many(
+        "stock.picking",
+        "batch_id",
+        states={
+            "draft": [("readonly", False)],
+            "waiting": [("readonly", False)],
+            "ready": [("readonly", False)],
+            "in_progress": [("readonly", False)],
+        },
+    )
+
+    @api.constrains("user_id")
```

Comment:
Did we agree on this api.depends or constraint

Reply:
As it is now, constraint.